src/components/with-datasets/redux/saga.ts

```diff
@@ -15,15 +15,33 @@ function* searchRequested({
   try {
     const auth = yield call(authService.getAuthorizationHeader);
 
-    const { data } = yield call(
-      axios.post,
-      `${getConfig().registrationApi.host}/search`,
-      searchRequest,
-      { headers: { Authorization: auth } }
-    );
-
-    if (data?.datasets) {
-      yield put(actions.searchDatasetsSucceeded(data.datasets as Dataset[]));
+    const datasets: Dataset[] = [];
+
+    if (searchRequest.queryRegistration) {
+      const { data: registrationSearch } = yield call(
+        axios.post,
+        `${getConfig().registrationApi.host}/search`,
+        searchRequest,
+        { headers: { Authorization: auth } }
+      );
+      if (registrationSearch?.datasets) {
+        datasets.push(...registrationSearch?.datasets);
+      }
+    }
+
+    if (searchRequest.queryFulltextSearch) {
+      const { data: fulltextSearch } = yield call(
+        axios.get,
+        'https://search.staging.fellesdatakatalog.digdir.no/suggestion/datasets',
+        { params: { q: searchRequest.query } }
+      );
+      if (fulltextSearch?.suggestions) {
+        datasets.push(...fulltextSearch?.suggestions);
+      }
+    }
```

Comment:
Her bør du prøve å bruke litt mer saga magi. Se denne for inspirasjon: https://github.com/Informasjonsforvaltning/dataset-catalog-gui/blob/55833912038f618f6c859ed613e4855f0bf8029c/src/components/with-enhetsregisteret/redux/saga.ts#L50